src/ngResource/resource.js

```diff
@@ -640,7 +642,7 @@ angular.module('ngResource', ['ng']).
         };
 
         forEach(actions, function(action, name) {
-          var hasBody = /^(POST|PUT|PATCH)$/i.test(action.method);
+          var hasBody = /^(POST|PUT|PATCH)$/i.test(action.method) || action.hasBody === true;
```

Comment:
I would find something like the following more intuitive:

```js
var hasBody = action.hasBody === true || (action.hasBody !== false && /^(POST|PUT|PATCH)$/i.test(action.method));
```

Am I missing any reason why we shouldn't allow people to overwrite `hasBody` for POST, PUT, PATCH? (Not that they should.)